UI Design Implementation

TimPurdum opened the pull request:
1. Implemented designs from @chantelle-campbell, with feedback from @codergrl. 
2. Added a Search feature to filter pages. Auto-changes on text change, and if there is only one match, hitting "Return" will navigate to that page.
3. Cleaned up layout of `ProjectionTool.razor` for mobile.

seahro commented on samples/dymaptic.GeoBlazor.Core.Sample.Shared/Shared/MainLayout.razor:
```diff
@@ -14,32 +14,37 @@
-                <label>Enter ArcGIS API Key:<input id="api-key-field" type="text" @bind="ApiKey" /></label>
-                <button id="api-key-go" @onclick="StateHasChanged">GO</button>
-                if (!string.IsNullOrWhiteSpace(clientId))
+            <img src="_content/dymaptic.GeoBlazor.Core.Sample.Shared/images/geoblazor_logo_with_title.png"
```
Everything Looks good

seahro reviewed (approved):
Nice visual changes.  Only style changes, radio buttons, hyperlinks and Main layout page.

TimPurdum commented:
layer list widget corresponds to gissue #40 .

TimPurdum commented:
@Chantelle Campbell just FYI, this task should still be open, but we had an issue with branch names that caused the syncing from GitHub to move it.

TimPurdum commented:
When merging in the #40 changes, we discovered a few things that needed to be updated here. @seahro and I pair-reviewed and implemented the following changes:
1. Updated `LayerLists.razor` for mobile screens.
2. Corrected the event handlers to be `ListItemCreated` instead of `ListItemSelected`.
3. Corrected a few bugs in the hookup system for `ListItemCreated`.
4. Implemented a basic example of setting visibility and titles in `LayerLists.razor`.
5. Added documentation (some sample xml docs for future reference to expand later, and some explaining unfinished advanced features).
